packages/contracts/src/resources/site.ts

```diff
@@ -713,13 +714,23 @@ export const siteApi = contract.router(
 					"Get a list of pubs by ID. This endpoint is used by the PubPub site builder to get a list of pubs.",
 				query: getPubQuerySchema
 					.extend({
+						pubIds: z
+							.array(pubsIdSchema)
+							.or(pubsIdSchema.transform((id) => [id]))
+							.optional()
+							.describe("Filter by pub ID."),
 						pubTypeId: pubTypesIdSchema
 							.array()
 							// this is necessary bc the query parser doesn't handle single string values as arrays
 							.or(pubTypesIdSchema.transform((id) => [id]))
 							.optional()
 							.describe("Filter by pub type IDs."),
-						stageId: stagesIdSchema.optional().describe("Filter by stage ID."),
+						stageId: stageConstraintSchema
+							.array()
+							// this is necessary bc the query parser doesn't handle single string values as arrays
+							.or(stagesIdSchema.transform((id) => [id]))
+							.optional()
+							.describe("Filter by stage ID."),
```

Comment:
kept the name "pubTypeId" and "stageId" to not force a break, but introduces "pubIds" bc it's technically a new option for this endpoint